src/porepy/models/fluid_mass_balance.py

```diff
@@ -412,11 +423,30 @@ def bc_values_darcy(self, subdomains: Sequence[pp.Grid]) -> pp.ad.Operator:
             Operator representing the boundary condition values for the Darcy flux.
 
         """
-        boundary_grids = self.mdg.subdomains_to_boundary_grids(subdomains)
+        boundary_grids = self.subdomains_to_boundary_grids(subdomains)
         pressure_dirichlet = self.pressure(boundary_grids)
         flux_neumann = self.darcy_flux(boundary_grids)
         return pressure_dirichlet + flux_neumann
 
+    def update_boundary_conditions(self, initial: bool) -> None:
+        """Set values for the pressure and the darcy flux on boundaries."""
+        super().update_boundary_conditions(initial=initial)
+
+        def pressure(boundary_grids: Sequence[pp.BoundaryGrid]) -> np.ndarray:
```

Comment:
If we define these functions outside as e.g. _boundary_pressure(...), we can reuse the outer function and tailor the inner ones in a run script.

Reply:
Agreed, changing it